File: FCMPushNotice801/app/src/main/java/com/example/fcmpushnotice801/MainActivity.java
package com.example.fcmpushnotice801;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.messaging.FirebaseMessaging;

public class MainActivity extends AppCompatActivity {

    public static final String TAG = "MainActivity";
    private TextView resultTextView(){
        return findViewById(R.id.resultTextView);
    }
    private TextView firebaseToken(){
        return findViewById(R.id.firebaseTokenText);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // 화면 구성이 끝난 후에 메소드를 호출함
        initFirebase();
    }

    // 토큰 값을 얻어옴
    public void initFirebase(){
        FirebaseMessaging.getInstance().getToken()
                .addOnCompleteListener(new OnCompleteListener<String>() {
                    @Override
                    public void onComplete(@NonNull Task<String> task) {
                        if (task.isSuccessful()) {
                            // Get new FCM registration token
                            String token = task.getResult();
                            firebaseToken().setText(token);
                        }
                    }
                });
    }
}

File: KTFCMKEY801/app/src/main/java/com/example/ktfcmkey801/MainActivity.kt
package com.example.ktfcmkey801

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import com.google.firebase.messaging.FirebaseMessaging

// ":" > 상속
class MainActivity : AppCompatActivity() {

    private val resultTextView: TextView by lazy {
        findViewById(R.id.resultTextView)
    }

    private val firebaseToken: TextView by lazy {
        findViewById(R.id.firebaseTokenText)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.i("태그", "생성")
        initFirebase()
    }

    // 토큰 값 받아오기
    private fun initFirebase(){
        // getInstance > 싱글톤
        FirebaseMessaging.getInstance().token
            .addOnCompleteListener{ task ->
                if(task.isSuccessful){
                    firebaseToken.text = task.result
                }
                else {
                    Log.i("MainActivity", "출력출력")
                }
            }
    }
}